File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/55-JumpGame/55-JumpGame.java
// Last updated: 8/4/2026, 7:24:24 PM
class Solution {
    public boolean canJump(int[] nums) {
        int n = nums.length;
        int maxi = 0;

        for(int i = 0 ; i < n; i++){
            if(i > maxi) return false;

            maxi = Math.max(i+nums[i] , maxi);

            if(maxi >= n) return true;
        }
        return true;
    }
}

File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/55-JumpGame/55-JumpGame.cpp
// Last updated: 8/4/2026, 7:24:19 PM
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int n = nums.size();
        int maxIdx = 0;

        for(int i = 0 ; i < n ; i++){
            if(i > maxIdx) return false;

            maxIdx = max(maxIdx , i + nums[i]);
            if(i+nums[i] >= n) return true;


        }
        return true;
    }
};